cashier-sample-google-play-billing/src/main/java/com/getkeepsafe/cashier/sample/googleplaybilling/MainActivity.java

```diff
@@ -0,0 +1,326 @@
+                    .setMessage("Product details:\n"+
+                                "Name: "+product.name()+"\n"+
+                                "SKU: "+product.sku()+"\n"+
+                                "Price: "+product.price()+"\n"+
+                                "Is sub: "+product.isSubscription()+"\n"+
+                                "Currency: "+product.currency())
+                    .setPositiveButton("OK", null)
+                    .create()
+                    .show();
+
+        }
+
+        @Override
+        public void failure(Vendor.Error error) {
+            Toast.makeText(MainActivity.this, "Product details failure!", Toast.LENGTH_SHORT).show();
+        }
+    };
+
+    @Override
+    protected void onCreate(Bundle savedInstance) {
+        super.onCreate(savedInstance);
+        setContentView(R.layout.activity_main);
+        useFake = (Switch) findViewById(R.id.use_fake);
+        final Button queryPurchases = (Button) findViewById(R.id.query_purchases);
+
+        itemsAdapter = new ItemsAdapter(this);
+
+        recyclerView = (RecyclerView)findViewById(R.id.items_recycler);
+        recyclerView.setAdapter(itemsAdapter);
+
+        progressBar = (ProgressBar)findViewById(R.id.items_progress);
+        progressBar.setVisibility(View.GONE);
+
+        itemsAdapter.setItemListener(new ItemsAdapter.ItemListener() {
+            @Override
+            public void onItemBuy(Item item) {
+                cashier.purchase(MainActivity.this, item.product, purchaseListener);
+            }
+
+            @Override
+            public void onItemUse(final Item item) {
+                progressDialog = new ProgressDialog(MainActivity.this);
+                progressDialog.setIndeterminate(true);
+                progressDialog.setTitle("Consuming item, please wait...");
+                progressDialog.show();
+                new Thread(new Runnable() {
+                    @Override
+                    public void run() {
+                        cashier.consume(item.purchase, consumeListener);
+                    }
+                }).start();
+            }
+
+            @Override
+            public void onItemGetDetails(Item item) {
+                cashier.getProductDetails(item.product.sku(), item.isSubscription, productDetailsListener);
+            }
+        });
+
+        testProduct = Product.create(
+                GooglePlayBillingConstants.VENDOR_PACKAGE,
+                "android.test.purchased",
+                "$0.99",
+                "USD",
+                "Test product",
+                "This is a test product",
+                false,
+                990_000L);
+
+        testProduct2 = Product.create(
+                GooglePlayBillingConstants.VENDOR_PACKAGE,
+                "com.abc.def.123",
+                "$123.99",
+                "USD",
+                "Test product 2",
+                "This is another test product",
+                false,
```

Comment:
I think it would be good to add a subscription product in the sample.

Reply:
I think these products are for the Fake Billing Vendor, so we can define them without developer console access.

Reply:
The hardcoded product definitions serve two purposes:
1. Define SKUs to query product info from Google
2. Define fake product info to use with fake vendor api.
When working with real api you need to have product defined in Google console, that's why the sample returns only one product for real vendor and two for fake - only one of those hardcoded products is actually defined in Google console - somewhere ...
Of course I can add subscription in the code, but it will work only with fake api. I think it would be confusing example.

Reply:
I do not see why there would be confusion since the fake billing vendor already return items not actually defined in the Google console. My thought was that, in the sample, we can show the different product types (sub vs non-sub) to make the user explicitly aware that such types exist.

You are right in saying that we should be able to define those within the Google Play Console for the actual Google vendor as well. I'll checkout where that in-app product is defined in the Google console once we are in the process of merging this into `master`.

This is a none blocker for internal usage, I'll keep it here as a reminder. Thanks.

Reply:
I found where the test products are defined, kindly check [this](https://developer.android.com/google/play/billing/billing_testing#solo). 

So there are defined static `SKU`s within the billing client for testing several scenarios. Since this is the case, it would not be then possible to showcase all possible cases for item type at least for the actual Google vendor, unless a published app is configured with in-app products.

With this, do you think it would be worth it to have these product sku's as part of the actual Google vendor in the sample? Then in the fake one have different item types? This if for completeness of the sample in terms of types of products and test scenarios.